test/jest/roktManager.spec.ts

```diff
@@ -0,0 +1,119 @@
+import RoktManager, { IRoktLauncher, ISelectPlacementsOptions } from "../../src/roktManager";
+
+describe('RoktManager', () => {
+    let roktManager: RoktManager;
+
+    beforeEach(() => {
+        roktManager = new RoktManager();
+    });
+
+    describe('constructor', () => {
+        it('should be initialized', () => {
+            expect(roktManager).toBeDefined();
+        });
+        it('should have a null launcher', () => {
+            expect(roktManager['launcher']).toBeNull();
+        });
+    });
+
+    describe('#attachLauncher', () => {
+        it('should attach a launcher', () => {
+            const launcher = {} as IRoktLauncher;
+            roktManager.attachLauncher(launcher);
+            expect(roktManager['launcher']).not.toBeNull();
+        });
+
+        it('should queue the launcher method if no launcher is attached', () => {
+            roktManager.attachLauncher(null);
+            expect(roktManager['launcher']).toBeNull();
+            expect(roktManager['messageQueue'].length).toBe(1);
+            expect(roktManager['messageQueue'][0].methodName).toBe('attachLauncher');
+            expect(roktManager['messageQueue'][0].payload).toBe(null);
+        });
+
+        it('should process the message queue if a launcher is attached', () => {
+            const launcher = jest.fn() as unknown as IRoktLauncher;
+
+            roktManager.attachLauncher(null);
+            expect(roktManager['launcher']).toBeNull();
+            expect(roktManager['messageQueue'].length).toBe(1);
+            expect(roktManager['messageQueue'][0].methodName).toBe('attachLauncher');
+            expect(roktManager['messageQueue'][0].payload).toBe(null);
+
+            roktManager.attachLauncher(launcher);
+            expect(roktManager['launcher']).not.toBeNull();
+            expect(roktManager['messageQueue'].length).toBe(0);
+        });
+    });
+
+    describe('#selectPlacements', () => {
+        it('should call selectPlacements', () => {
+            const launcher = {
+                selectPlacements: jest.fn()
+            } as unknown as IRoktLauncher;
```

Comment:
I don't think there's a reason to make it `unknown`

```suggestion
            const launcher: IRoktLauncher = {
                selectPlacements: jest.fn()
            }
```